Assets/MixedRealityToolkit/_Core/Devices/MixedReality/WMRDevice.cs

```diff
@@ -0,0 +1,489 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Internal.Definitions;
+using Microsoft.MixedReality.Toolkit.Internal.Extensions;
+using Microsoft.MixedReality.Toolkit.Internal.Interfaces.InputSystem;
+using Microsoft.MixedReality.Toolkit.Internal.Managers;
+using Microsoft.MixedReality.Toolkit.Internal.Utilities;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.XR.WSA.Input;
+
+namespace Microsoft.MixedReality.Toolkit.Internal.Devices.WindowsMixedReality
+{
+    public class WMRDevice : IMixedRealityDevice
+    {
+        private readonly Dictionary<uint, IMixedRealityInputSource> activeControllers = new Dictionary<uint, IMixedRealityInputSource>();
+        Vector3 controllerPosition, pointerPosition, gripPosition = Vector3.zero;
+        Quaternion controllerRotation, pointerRotation, gripRotation = Quaternion.identity;
```

Comment:
nit, let's declare our fields on separate lines, and use access modifiers.

Reply:
tbc, when it works :P